app/src/main/java/org/wikipedia/util/L10nUtil.kt

```diff
@@ -0,0 +1,129 @@
+        val dir = Character.getDirectionality(c).toInt()
+        return dir == Character.DIRECTIONALITY_RIGHT_TO_LEFT.toInt() || dir == Character.DIRECTIONALITY_RIGHT_TO_LEFT_ARABIC.toInt()
+    }
+
+    @JvmStatic
+    fun getStringForArticleLanguage(languageCode: String, resId: Int): String {
+        return getStringsForLocale(Locale(languageCode), intArrayOf(resId))[resId]
+    }
+
+    @JvmStatic
+    fun getStringForArticleLanguage(title: PageTitle, resId: Int): String {
+        return getStringsForLocale(Locale(title.wikiSite.languageCode()), intArrayOf(resId))[resId]
+    }
+
+    fun getStringsForArticleLanguage(title: PageTitle, resId: IntArray): SparseArray<String> {
+        return getStringsForLocale(Locale(title.wikiSite.languageCode()), resId)
+    }
+
+    private fun getStringsForLocale(targetLocale: Locale,
+                                    @StringRes strings: IntArray): SparseArray<String> {
+        val config = currentConfiguration
+        val systemLocale = ConfigurationCompat.getLocales(config)[0]
+        if (systemLocale.language == targetLocale.language) {
+            val localizedStrings = SparseArray<String>()
+            for (stringRes in strings) {
+                localizedStrings.put(stringRes, WikipediaApp.getInstance().getString(stringRes))
+            }
+            return localizedStrings
+        }
+        setDesiredLocale(config, targetLocale)
+        val localizedStrings = getTargetStrings(strings, config)
+        config.setLocale(systemLocale)
+        resetConfiguration(config)
+        return localizedStrings
+    }
+
+    private val currentConfiguration: Configuration
+        get() = Configuration(WikipediaApp.getInstance().resources.configuration)
+
+    private fun getTargetStrings(@StringRes strings: IntArray, altConfig: Configuration): SparseArray<String> {
+        val localizedStrings = SparseArray<String>()
+        val targetResources = Resources(WikipediaApp.getInstance().resources.assets,
+                WikipediaApp.getInstance().resources.displayMetrics,
+                altConfig)
+        for (stringRes in strings) {
+            localizedStrings.put(stringRes, targetResources.getString(stringRes))
+        }
+        return localizedStrings
+    }
+
+    private fun resetConfiguration(defaultConfig: Configuration) {
+        Resources(WikipediaApp.getInstance().resources.assets,
+                WikipediaApp.getInstance().resources.displayMetrics,
+                defaultConfig)
+    }
+
+    private fun getDesiredLocale(desiredLocale: Locale): Locale {
+        // TODO: maybe other language variants also have this issue, we need to add manually. e.g. kk?
+        return when (desiredLocale.language) {
+            AppLanguageLookUpTable.TRADITIONAL_CHINESE_LANGUAGE_CODE, AppLanguageLookUpTable.CHINESE_TW_LANGUAGE_CODE, AppLanguageLookUpTable.CHINESE_HK_LANGUAGE_CODE, AppLanguageLookUpTable.CHINESE_MO_LANGUAGE_CODE -> Locale.TRADITIONAL_CHINESE
```

Comment:
Would it be possible to import `AppLanguageLookUpTable` to reduce the length of this line?